### PR TITLE
njalla: fix record id unmarshal error

### LINKED ISSUE #1685: Njalla dns provider not working anymore: unmarshal error
### Welcome

- [X] Yes, I'm using a binary release within 2 latest releases.
- [X] Yes, I've searched similar issues on GitHub and didn't find any.
- [X] Yes, I've included all information below (version, config, etc).

### What did you expect to see?

That dnsChallenge using traefik and the dns provider 'njalla' returns valid certificates.

Additional notes:
- Also reproducible using docker image without traefik, therefor I included the reproduction steps of the docker image.
- It did work a few weeks / maybe a month ago, so maybe the API changed?

### What did you see instead?

acme: error presenting token: njalla: failed to add record: failed to unmarshal response result: json: cannot unmarshal string into Go struct field Record.id of type int

### How do you use lego?

Docker image

### Reproduction steps

1. docker run -it -e NJALLA_TOKEN=51b...2e7 goacme/lego:v4.8.0 --email "development@manu.software" --dns njalla --domains "test.manu.software" -a run

### Version of lego

```console
lego version v4.8.0 linux/amd64
```

### Logs

<details>

```console
2022/08/01 18:34:56 No key found for account development@manu.software. Generating a P256 key.
2022/08/01 18:34:56 Saved key to /.lego/accounts/acme-v02.api.letsencrypt.org/development@manu.software/keys/development@manu.software.key
2022/08/01 18:34:57 [INFO] acme: Registering account for development@manu.software
!!!! HEADS UP !!!!

Your account credentials have been saved in your Let's Encrypt
configuration directory at "/.lego/accounts".

You should make a secure backup of this folder now. This
configuration directory will also contain certificates and
private keys obtained from Let's Encrypt so making regular
backups of this folder is ideal.
2022/08/01 18:34:58 [INFO] [test.manu.software] acme: Obtaining bundled SAN certificate
2022/08/01 18:34:59 [INFO] [test.manu.software] AuthURL: https://acme-v02.api.letsencrypt.org/acme/authz-v3/137177728806
2022/08/01 18:34:59 [INFO] [test.manu.software] acme: Could not find solver for: tls-alpn-01
2022/08/01 18:34:59 [INFO] [test.manu.software] acme: Could not find solver for: http-01
2022/08/01 18:34:59 [INFO] [test.manu.software] acme: use dns-01 solver
2022/08/01 18:34:59 [INFO] [test.manu.software] acme: Preparing to solve DNS-01
2022/08/01 18:34:59 [INFO] [test.manu.software] acme: Cleaning DNS-01 challenge
2022/08/01 18:34:59 [WARN] [test.manu.software] acme: cleaning up failed: njalla: unknown record ID for '_acme-challenge.test.manu.software.' 'cEmnxvprPUKZNbXWLHXcO5YWS3ELCjWjIucSHAIPP78'
2022/08/01 18:34:59 [INFO] Deactivating auth: https://acme-v02.api.letsencrypt.org/acme/authz-v3/137177728806
2022/08/01 18:35:00 Could not obtain certificates:
	error: one or more domains had a problem:
[test.manu.software] [test.manu.software] acme: error presenting token: njalla: failed to add record: failed to unmarshal response result: json: cannot unmarshal string into Go struct field Record.id of type int
```

</details>

### Go environment (if applicable)

<details>

```console
$ go version && go env
# paste output here
```

</details>

### PR DESCRIPTION
**Pull request by manuel2258**
fixes #1685

### CODE REVIEW

File: providers/dns/njalla/internal/client.go [hunk 1]
```diff
@@ -52,7 +53,7 @@ func (c *Client) RemoveRecord(id int, domain string) error {
 	data := APIRequest{
 		Method: "remove-record",
 		Params: Record{
-			ID:     id,
+			ID:     strconv.Itoa(id),
```
Comment (manuel2258) [inline providers/dns/njalla/internal/client.go:56]: Is usage of strconv fine or is there something better to use here?

**Comment by ldez** [comment]
Can you run the following command and put the output of this command to a comment in your PR:
```bash
rm -rf .lego

./lego -m your@email.com --dns YOUR_PROVIDER_NAME -d *.example.com -d example.com -s https://acme-staging-v02.api.letsencrypt.org/directory run
```
Note the wildcard domain is important.

File: providers/dns/njalla/internal/client.go (hunk 1 above)
Comment (ldez) [reply providers/dns/njalla/internal/client.go:56]: from the extract of the doc, I think that all the ID must be a `int`

Comment (manuel2258) [reply providers/dns/njalla/internal/client.go:56]: yeah that exactly is the problem, they now return an int boxed as a string instead of an int

Comment (ldez) [reply providers/dns/njalla/internal/client.go:56]: so I don't understand why you have to convert a string to int here

Comment (manuel2258) [reply providers/dns/njalla/internal/client.go]: I though that the ID is part of the interface but its not, I removed the conversions now

**Comment by manuel2258** [comment]
I'm unsure whether that's to be expected but running against the staging letsencrypt server fails (I guess I was supposed to fill in the example values, right?):

```
$ NJALLA_TOKEN=51b...2e7 ./dist/lego -m "development@manu.software" --dns njalla -d "*.oecis.io" -d "oecis.io" -s https://acme-staging-v02.api.letsencrypt.org/directory -a run
2022/08/02 19:51:16 No key found for account development@manu.software. Generating a P256 key.
2022/08/02 19:51:16 Saved key to /home/manuel/projects/lego/.lego/accounts/acme-staging-v02.api.letsencrypt.org/development@manu.software/keys/development@manu.software.key
2022/08/02 19:51:17 [INFO] acme: Registering account for development@manu.software
!!!! HEADS UP !!!!

Your account credentials have been saved in your Let's Encrypt
configuration directory at "/home/manuel/projects/lego/.lego/accounts".

You should make a secure backup of this folder now. This
configuration directory will also contain certificates and
private keys obtained from Let's Encrypt so making regular
backups of this folder is ideal.
2022/08/02 19:51:17 [INFO] [*.oecis.io, oecis.io] acme: Obtaining bundled SAN certificate
2022/08/02 19:51:18 [INFO] [*.oecis.io] AuthURL: https://acme-staging-v02.api.letsencrypt.org/acme/authz-v3/3212330674
2022/08/02 19:51:18 [INFO] [oecis.io] AuthURL: https://acme-staging-v02.api.letsencrypt.org/acme/authz-v3/3212330684
2022/08/02 19:51:18 [INFO] [*.oecis.io] acme: use dns-01 solver
2022/08/02 19:51:18 [INFO] [oecis.io] acme: Could not find solver for: tls-alpn-01
2022/08/02 19:51:18 [INFO] [oecis.io] acme: Could not find solver for: http-01
2022/08/02 19:51:18 [INFO] [oecis.io] acme: use dns-01 solver
2022/08/02 19:51:18 [INFO] [*.oecis.io] acme: Preparing to solve DNS-01
2022/08/02 19:51:19 [INFO] [oecis.io] acme: Preparing to solve DNS-01
2022/08/02 19:51:19 [INFO] [*.oecis.io] acme: Trying to solve DNS-01
2022/08/02 19:51:19 [INFO] [*.oecis.io] acme: Checking DNS record propagation using [127.0.0.53:53]
2022/08/02 19:51:21 [INFO] Wait for propagation [timeout: 1m0s, interval: 2s]
2022/08/02 19:51:21 [INFO] [*.oecis.io] acme: Waiting for DNS record propagation.
2022/08/02 19:51:27 [INFO] [oecis.io] acme: Trying to solve DNS-01
2022/08/02 19:51:27 [INFO] [oecis.io] acme: Checking DNS record propagation using [127.0.0.53:53]
2022/08/02 19:51:29 [INFO] Wait for propagation [timeout: 1m0s, interval: 2s]
2022/08/02 19:51:29 [INFO] [oecis.io] The server validated our request
2022/08/02 19:51:29 [INFO] [*.oecis.io] acme: Cleaning DNS-01 challenge
2022/08/02 19:51:30 [INFO] [oecis.io] acme: Cleaning DNS-01 challenge
2022/08/02 19:51:30 [INFO] Deactivating auth: https://acme-staging-v02.api.letsencrypt.org/acme/authz-v3/3212330674
2022/08/02 19:51:31 [INFO] Skipping deactivating of valid auth: https://acme-staging-v02.api.letsencrypt.org/acme/authz-v3/3212330684
2022/08/02 19:51:31 Could not obtain certificates:
	error: one or more domains had a problem:
[*.oecis.io] acme: error: 400 :: urn:ietf:params:acme:error:dns :: During secondary validation: DNS problem: SERVFAIL looking up TXT for _acme-challenge.oecis.io - the domain's nameservers may be malfunctioning
```

against the non staging server it seems to work fine:

```
$ NJALLA_TOKEN=51b...2e7 ./dist/lego -m "development@manu.software" --dns njalla -d "*.oecis.io" -d "oecis.io" -a run
2022/08/02 19:50:27 No key found for account development@manu.software. Generating a P256 key.
2022/08/02 19:50:27 Saved key to /home/manuel/projects/lego/.lego/accounts/acme-v02.api.letsencrypt.org/development@manu.software/keys/development@manu.software.key
2022/08/02 19:50:28 [INFO] acme: Registering account for development@manu.software
!!!! HEADS UP !!!!

Your account credentials have been saved in your Let's Encrypt
configuration directory at "/home/manuel/projects/lego/.lego/accounts".

You should make a secure backup of this folder now. This
configuration directory will also contain certificates and
private keys obtained from Let's Encrypt so making regular
backups of this folder is ideal.
2022/08/02 19:50:28 [INFO] [*.oecis.io, oecis.io] acme: Obtaining bundled SAN certificate
2022/08/02 19:50:30 [INFO] [*.oecis.io] AuthURL: https://acme-v02.api.letsencrypt.org/acme/authz-v3/137553704406
2022/08/02 19:50:30 [INFO] [oecis.io] AuthURL: https://acme-v02.api.letsencrypt.org/acme/authz-v3/137553704416
2022/08/02 19:50:30 [INFO] [*.oecis.io] acme: use dns-01 solver
2022/08/02 19:50:30 [INFO] [oecis.io] acme: Could not find solver for: tls-alpn-01
2022/08/02 19:50:30 [INFO] [oecis.io] acme: Could not find solver for: http-01
2022/08/02 19:50:30 [INFO] [oecis.io] acme: use dns-01 solver
2022/08/02 19:50:30 [INFO] [*.oecis.io] acme: Preparing to solve DNS-01
2022/08/02 19:50:30 [INFO] [oecis.io] acme: Preparing to solve DNS-01
2022/08/02 19:50:30 [INFO] [*.oecis.io] acme: Trying to solve DNS-01
2022/08/02 19:50:30 [INFO] [*.oecis.io] acme: Checking DNS record propagation using [127.0.0.53:53]
2022/08/02 19:50:32 [INFO] Wait for propagation [timeout: 1m0s, interval: 2s]
2022/08/02 19:50:32 [INFO] [*.oecis.io] acme: Waiting for DNS record propagation.
2022/08/02 19:50:39 [INFO] [*.oecis.io] The server validated our request
2022/08/02 19:50:39 [INFO] [oecis.io] acme: Trying to solve DNS-01
2022/08/02 19:50:39 [INFO] [oecis.io] acme: Checking DNS record propagation using [127.0.0.53:53]
2022/08/02 19:50:41 [INFO] Wait for propagation [timeout: 1m0s, interval: 2s]
2022/08/02 19:50:45 [INFO] [oecis.io] The server validated our request
2022/08/02 19:50:45 [INFO] [*.oecis.io] acme: Cleaning DNS-01 challenge
2022/08/02 19:50:46 [INFO] [oecis.io] acme: Cleaning DNS-01 challenge
2022/08/02 19:50:46 [INFO] [*.oecis.io, oecis.io] acme: Validations succeeded; requesting certificates
2022/08/02 19:50:48 [INFO] [*.oecis.io] Server responded with a certificate.
```

**Comment by ldez** [comment]
Just a tip: address changes as commit instead of squash+push-force.
Because it's impossible for me to see changes when you squash.

**Review by ldez — APPROVED** [review]
LGTM

**Pull request merged**